das-query-engine/validators/actions.py

```diff
@@ -53,28 +53,31 @@ class QueryValidator(PayloadValidator):
     strict = True
 
     @staticmethod
-    def validate_query(query, *args, **kwargs) -> bool:
-        if not isinstance(query, dict):
+    def validate_query(queries, *args, **kwargs) -> bool:
+        if isinstance(queries, dict):
+            queries = [queries]
+        elif not isinstance(queries, list) or not all(isinstance(query, dict) for query in queries):
```

Comment:
This limits the query, such as `[{}, []] `or `[[], []]`. To ensure the parameters are correct, we can to perform a recursive check and run additional tests, or simply verify the top-level type as either a list or a dictionary.

Reply:
> simply verify the top-level type as either a list or a dictionary.

This sounds like a good plan for now. @angeloprobst , do you have any inputs here?

Reply:
I think that a recursive validation like the one suggested by @Pedrobc89 in the [ticket](https://github.com/singnet/das-serverless-functions/issues/156) covers all scenarios.